.github/workflows/build-and-publish.yml

```diff
@@ -33,7 +33,26 @@ on:
         type: string
 
 jobs:
+  # check ECR image job here
+  # see: https://github.com/navapbc/archive-massgov-pfml/blob/5381834193ac9da8066180a8ab1d0589b014d26b/.github/workflows/api-deploy.yml#L319-L324
+  # Skip if its there
+  # make release-image-tag
+  check-ecr-image:
+    name: Check if ECR image has been created
+    runs-on: ubuntu-latest
+
+    steps: 
+      - uses: actions/checkout@v3
+
+      - name: Check ECR Image
+        id: image-tag
+        run: |
+          IMAGE_TAG=$(make release-image-tag)
+          aws ecr describe-images --repository-name=CHANGE_THIS--image-ids=imageTag=$IMAGE_TAG --region us-east-1
+
   build-and-publish:
+    depends-on: [check-ecr-image]
+    # TODO find a way to reference the above
```

Comment:
I don't think `depends-on` is the logic you want here. What you're probably looking for is:

```
check-ecr-image() || make release-build
```

Which resolves to

```
when check-ecr-image() is false || then run  make release-build
when check-ecr-image() is true  || then skip make release-build
```

Reply:
Actually, checking that the ECR image is published is superfluous given that when are moving the build and publish step to a place that only runs once